indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -573,7 +580,8 @@ private Map<String, TaskReport> getTaskCompletionReports()
                 getTaskCompletionRowStats(),
                 errorMsg,
                 segmentAvailabilityConfirmationCompleted,
-                segmentAvailabilityWaitTimeMs
+                segmentAvailabilityWaitTimeMs,
+                JodaUtils.condenseIntervals(intervals)
```

Comment:
Hi, thanks for taking a look at this PR!

Making `ingestedIntervals` an attribute of the Task doesn't feel right to me because this is a very transient thing.  The instance variable would be empty all the time except for a vanishingly small period of time after `generateAndPublishSegments()` completes and before the task is torn down.  We only pass intervals to `getTaskCompletionReports()` in one place - we would call the setter and then immediately call `getTaskCompletionReports()`, which would call the getter.  And no other entity has any interest in these intervals - they are not really a property of the Task.

Let me know what you think.

Reply:
Yeah, that makes sense. I just went with that choice to be consistent. This works as well.